node/test/toolrunnertests.ts

```diff
@@ -1254,6 +1254,43 @@ describe('Toolrunner Tests', function () {
                     done(err);
                 });
         });
+
+        // ----------------------------------------
+        // double quotes in arg (OSX/Linux)
+        // ----------------------------------------
+
+        it('exec with double quotes in args (OSX/Linux)', function (done) {
+            this.timeout(10000);
+            let bashPath = tl.which('bash');
+            let bash = tl.tool(bashPath)
+                .arg('-c')
+                .arg('echo $0')
+                .arg('hello"world"')
```

Comment:
this does not appear to be testing the issue when the secret was literally: 
hello\"secret

Reply:
The test is for the case when the secret is literally `hello\"secret`. (When the secret is `hello"secret`, the `"` gets eaten-up somewhere in between, and the value here is `helloworld`). This discussion is there in the conversation.